### DISCUSSION TITLE
remove website url and other lines on save

### DISCUSSION THREAD

**Discussion by stanleyblade** [atom/atom#22136] [Q&A]
When saving a file (specifically to the minified file) I would like to automatically remove all the url "https://websiteaddress.com/" references and other lines. Including removing the stylesheet line altogether along with the body css.

![Screenshot_3](https://user-images.githubusercontent.com/81980813/113597354-7c6dc800-9633-11eb-9f66-b2b3c4677f29.png)

How could I go about this? Thank you.

**Comment by UziTech** [comment]
This sounds like a question that should be asked on [stackoverflow.com](http://stackoverflow.com/) since it is not specific to Atom.

**Reply by ThatXliner** [reply]
Well, then there's the Atom TextEditor API you need for reading and writing text buffer contents. I think a simple regex may be useful for this situation...

**Reply by stanleyblade** [reply]
I had to look up what a regex was, im really fresh to anything but html and css, thanks for the reply it has put me on the right path i think

**Comment by UziTech** [comment]
You may be looking for something like [atom-minify](https://atom.io/packages/atom-minify)

**Reply by stanleyblade** [reply]
Hi UziTech, I already have the atom-minify to shrink the pages before loading onto the website, but its the search and replace certain text in pages automatically im looking for.

**Reply by UziTech** [reply]
<kbd>CTRL/CMD</kbd> + <kbd>F</kbd> Should allow you to find/replace text on the page. I don't exactly understand what you are trying to do. If you want things replaced when you minify it would depend on the software you are using to minify.

You will probably need to look at the [options](https://github.com/armin-pfaeffle/atom-minify/blob/master/OPTIONS.md#js-minifiers) for each minifier.

**Reply by stanleyblade** [reply]
I want to do find and replace for the website url (and other terms) and replace them automatically on save to the minified file.  Thanks for the link I shall have a look.

**Reply by stanleyblade** [reply]
simply put, this:
![Screenshot_7](https://user-images.githubusercontent.com/81980813/113624064-185cfb00-9657-11eb-8b59-ab05a536868e.png)

**Reply by UziTech** [reply]
That looks like something the minifier would do not Atom. The correct option would depend on whether you are using uglifyjs, YUI Compressor, or Closure Compiler.

This question might be answered better if you ask it on [atom-minify](https://github.com/armin-pfaeffle/atom-minify) or the specific minifier's repo.

**Reply by stanleyblade** [reply]
thank you for the link, i always manage to post in the wrong place!

**Comment by ThatXliner** [comment]
Use a regex (might break things though. use with caution):

Put this in Atom find (regex mode):
```regex
((?:href)|(?:src)=")https:\/\/websiteaddress\.com([^"]+")
```
And put this in replace:

```php
$1$2
```

test:
https://regex101.com/r/VAi3mS/1

**Reply by UziTech** [reply]
I think they want to replace the text in the minified file not the source file.

**Reply by ThatXliner** [reply]
I think it should be the same, minified or not. Why should the regex break on minified HTML?

**Reply by UziTech** [reply]
I don't think they want to run the find and replace every time they save the file. They want something that will do it automatically on save.

**Reply by ThatXliner** [reply]
What about adding a callback to editor:save?

**Reply by UziTech** [reply]
How do you do that?

**Reply by stanleyblade** [reply]
This is brilliant, thank you for the test link, learning alot at the moment! If only it would run the script on the save to minify that would be perfect,

I had an idea for an alternative option of creating a `.bat` file with a find and replace  `.min` files for the url reference. I think a `.bat` file will be easier for me to work out.

I am reaching too far beyond my capabilities with atom. Thank you for the help :)

**Reply by ThatXliner** [reply]
> This is brilliant, thank you for the test link, learning alot at the moment! If only it would run the script on the save to minify that would be perfect,

😄 

> I am reaching too far beyond my capabilities with atom. Thank you for the help :)

No problem! :)

> How do you do that?

(shove this in your init script. I haven't tested it yet but it should work)
```coffeescript
atom.workspace.observeTextEditors (editor) ->
    editor.onDidSave ->
        if editor.getPath().endsWith("html")  # Is an HTML file
            editor.setText(
                editor
                .getText()
                .scan(/((?:href)|(?:src)=")https:\/\/websiteaddress\.com([^"]+")/)
                .forEach(x -> x.replace("$1$2"))
            )
```

**Reply by UziTech** [reply]
It seems like that code is changing the source file not the minified file.

**Reply by ThatXliner** [reply]
Oh. Well, then you can replace the
```coffeescript
        if editor.getPath().endsWith("html")  # Is an HTML file
```
with
```coffeescript
        if editor.getPath().endsWith(".min.html")  # Is an HTML file
```

I guess. Or you can do in a [gulp file](https://gulpjs.com) or python script or whatnot.
